### PR TITLE
[Native] Default to dummy reporter when prometheus is disabled

### PR DESCRIPTION
**Pull request by karteekmurthys**
## Description

The DummyStatsReporter was disabled after introducing prometheus reporter for stats collection. With this
change, we reintroduce it and register as singleton when stats reporters are registered.

## Testing
Tested locally by starting the presto cpp server and fetching metrics on browser.

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by majetideepak** [comment]
@karteekmurthys looks like you are fixing the changes introduced here https://github.com/prestodb/presto/pull/23146
Can you add that PR's author as a co-author and make a note in that PR?

File: presto-native-execution/CMakeLists.txt
Comment (majetideepak) [inline presto-native-execution/CMakeLists.txt:61]: update CI as well?

Comment (karteekmurthys) [reply presto-native-execution/CMakeLists.txt:61]: done.

**Review by majetideepak — DISMISSED** [review]
changes look good to me.
@amitkdutta, @tanjialiang can you take a look as well?

**Comment by amitkdutta** [comment]
@tanjialiang Please take a look.

**Comment by tanjialiang** [comment]
Thank you @karteekmurthys for the fix. Looks good to me. Please make sure the following before landing:
1) the format check passes
2) local server runs without PROMETHEUS.

**Review by tanjialiang — DISMISSED** [review]

**Comment by karteekmurthys** [comment]
> Thank you @karteekmurthys for the fix. Looks good to me. Please make sure the following before landing:
> 
> 1. the format check passes
> 2. local server runs without PROMETHEUS.

1. Format check is a clang_format tool issue.
2. I have verfied locally the cluster comes up without Prometheus.

**Comment by majetideepak** [comment]
@karteekmurthys can you cherry-pick the commit here https://github.com/prestodb/presto/pull/23222 to verify the format-check passes? Thanks!

**Comment by karteekmurthys** [comment]
> @karteekmurthys can you cherry-pick the commit here #23222 to verify the format-check passes? Thanks!

It fixes the issue: https://app.circleci.com/pipelines/github/prestodb/presto/17936/workflows/e82eb3b0-4a47-437b-b2e8-bc850e9670f5/jobs/70935

**Comment by majetideepak** [comment]
@karteekmurthys Thanks! I will merge that PR and you can rebase after that.

**Comment by majetideepak** [comment]
@karteekmurthys can you rebase with master? Thanks!

**Comment by majetideepak** [comment]
@tdcmeehan: In https://github.com/prestodb/presto/pull/23222, I mistakenly removed the bits to advance Velox for the CI format-check job. Those bits are required for the Velox script. This was not caught since that PR did not trigger the check job.

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**
